concent_api/core/tests/test_unit_authentication_utilities.py

```diff
@@ -0,0 +1,214 @@
+            CONCENT_PRIVATE_KEY,
+            self.REQUESTOR_PUBLIC_KEY,
+        )
+
+        self.assertEqual(loaded_task_to_compute_with_utility_function, loaded_task_to_compute_with_golem_messages_load)
+
+        for slot in message.TaskToCompute.__slots__:
+            self.assertEqual(
+                getattr(loaded_task_to_compute_with_utility_function,       slot),
+                getattr(loaded_task_to_compute_with_golem_messages_load,    slot)
+            )
+
+
+class ValidateGolemMessageClientAuthorizationUnitTest(ConcentIntegrationTestCase):
+
+    def test_validate_golem_message_client_authorization_should_not_raise_error_if_correct_message_is_used(self):
+        client_authorization = message.concents.ClientAuthorization(
+            client_public_key = CONCENT_PUBLIC_KEY
+        )
+
+        try:
+            validate_golem_message_client_authorization(client_authorization)
+        except Exception:  # pylint: disable=broad-except
+            self.fail()
+
+    def test_validate_golem_message_client_authorization_should_raise_400_error_when_wrong_message_is_used(self):
+        ping = message.Ping()
+
+        with self.assertRaises(Http400):
+            validate_golem_message_client_authorization(ping)
+
+    def test_validate_golem_message_client_authorization_should_raise_400_error_when_public_key_is_not_string(self):
+        client_authorization = message.concents.ClientAuthorization(
+            client_public_key = 111
+        )
+
+        with self.assertRaises(Http400):
+            validate_golem_message_client_authorization(client_authorization)
+
+    def test_validate_golem_message_client_authorization_should_raise_400_error_when_public_key_is_not_bytes(self):
+        client_authorization = message.concents.ClientAuthorization(
+            client_public_key = 'key'
+        )
+
+        with self.assertRaises(Http400):
+            validate_golem_message_client_authorization(client_authorization)
+
+    def test_validate_golem_message_client_authorization_should_raise_400_error_when_public_key_length_is_wrong(self):
+        client_authorization = message.concents.ClientAuthorization(
+            client_public_key = CONCENT_PUBLIC_KEY[:-1]
+        )
+
+        with self.assertRaises(Http400):
+            validate_golem_message_client_authorization(client_authorization)
+
+
+class ValidateGolemMessageSignedWithKeyUnitTest(ConcentIntegrationTestCase):
+
+    def test_validate_golem_message_signed_with_key_should_not_raise_error_if_correct_message_and_key_is_used(self):
+        ping = message.Ping()
+
+        dumped_ping = dump(ping, CONCENT_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+        ping = load(dumped_ping, CONCENT_PRIVATE_KEY, CONCENT_PUBLIC_KEY)
+
+        assert ping.sig is not None
+
+        try:
+            validate_golem_message_signed_with_key(
+                ping,
+                CONCENT_PUBLIC_KEY,
+            )
+        except Exception:  # pylint: disable=broad-except
```

Comment:
Plase don't catch `Exception` unless there's absolutely no other way. I agree with pylint here.

An exception will make the test fail anyway so it's fine not to catch it. If you really want it to be a test failure (rather than an error), catch specific exception types that you know can be raised.

Stuff like `SyntaxError`, `TypeError`, etc, should always be let through unless you're writing a top-level catch-all handler that can be made comprehensive enough to sensily handle different kinds of errors.

Reply:
The test checks if the function doesn't raise any exception. If it does, the test fails. It is a different thing that when exception is raised by test - it results in error, not in fail, and gives a different output. 

In this case, I want to test if function was executed correctly. It is constructed this way, because function doesn't return any value which can be checked. If you know better construction for checking that case please let me know.

Reply:
I understand that the output is different but I don't think it's different enough to be a problem.

Basically, I think that it's better to catch too little than too much and for me this applies both to tests and to normal code. If we catch too little, there's a very high chance that we find out in testing and update the handler to include that particular exception type. If we catch too much, we might never know about the error. I think it's better to avoid an overly general handler and miss some exceptions because catching stuff like `SyntaxError` is worse.

I think that there's rarely a real need to catch and not immediately re-raise `Exception` or `BaseException`, especially in tests, and any such case should should be reviewed very thoughly because misuse can cause a lot of trouble.

I know that this particular code will fail anyway and we won't really overlook the error but the cause of the problem will be misrepresented and it may still be a minor problem in specific circumstances. For example you might think that after a refactoring a part of code is failing because it's just not ready yet and skip the test while it's actually a simple problem in the test itself. A problem that would be evident if exceptions were not being caught indiscriminately.

There's definitely an exception type you expect this code to raise if there's an error. Add this exception (or exceptions) to the handler and let everything else escape and just be treated as an error. If you find out that there's more that can be raised, update the handler when you notice. That's better than catching everything.